addons/hr_expense/static/src/views/kanban.xml

```diff
@@ -18,21 +18,40 @@
     </t>
 
     <t t-name="hr_expense.KanbanButtons" t-inherit="web.KanbanView.Buttons" t-inherit-mode="primary">
+        <xpath expr="//div[hasclass('o_cp_buttons')]" position="attributes">
+            <!-- Ensure that dropdown items show vertically and not side-by-side -->
+            <attribute name="class" remove="d-flex" separator=" "/>
+            <attribute name="class" add="d-block d-xl-flex" separator=" "/>
+        </xpath>
         <xpath expr="//div[hasclass('o_cp_buttons')]" position="inside">
             <input type="file" name="ufile" class="d-none" t-ref="fileInput" multiple="1" accept="*" t-on-change="onChangeFileInput" />
+            <!-- If Create Report is shown and window isSmall, 'New' button is added to the 'control-panel-additional-actions' slot (dropdown in mobile) -->
+            <t t-if="canCreate and props.showButtons and env.isSmall and displayCreateReport()">
+                <button type="button" class="btn btn-primary o-kanban-button-new" data-hotkey="c" t-on-click="() => this.createRecord()" data-bounce-button="">
+                    New
+                </button>
+            </t>
```

Comment:
Asking for a review because OCO is on holidays, and in this forward port I made a few changes compared to the previosu versions. The change affects the 'New' button to match the task requirements for this specific version for mobile: if 'Create Report' is shown, 'New' should be in the dropdown. Else, 'New' is displayed in its usual place (in the control-panel-create-button slot).

Reply:
LGTM